consensus/XDPoS/engines/engine_v2/epochSwitch.go

```diff
@@ -40,9 +40,10 @@ func (x *XDPoS_v2) getEpochSwitchInfo(chain consensus.ChainReader, header *types
 			penalties := []common.Address{}
 			standbynodes := []common.Address{}
 			epochSwitchInfo := &types.EpochSwitchInfo{
-				Penalties:    penalties,
-				Standbynodes: standbynodes,
-				Masternodes:  masternodes,
+				Penalties:      penalties,
+				Standbynodes:   standbynodes,
+				Masternodes:    masternodes,
+				MasternodesLen: len(masternodes),
```

Comment:
hmm... why this need to be added?

Reply:
This change is for dynamic masternodes

This line is specific for block 0 which I didn't include in the first merge fix.